### PR TITLE
(abandoned) Move labels to ep

### PR DESCRIPTION
**Pull request by amreshakim**
Decouple labels from containers and store them in endpoints.
- change api to allow passing labels during endpoint creation
- regenerate swagger API
- remove labels from containers
- store labels in endpoints

https://github.com/cilium/cilium/issues/1041

### CODE REVIEW

**Comment by aanm** [comment]
@amreshakim I've marked this PR with the `WIP` label, ping me once you're done or remove and add the reviewers to review it.

File: daemon/docker_watcher.go [hunk 1]
```diff
@@ -238,13 +238,8 @@ func (d *Daemon) handleCreateContainer(id string, retry bool) {
...
 				ep = endpointmanager.LookupCiliumIDLocked(cid)
-				if ep != nil {
-					// Associate container id with endpoint
-					ep.Mutex.Lock()
-					ep.DockerID = id
-					ep.Mutex.Unlock()
-					endpointmanager.LinkContainerID(ep)
-				}
+				// Associate container id with endpoint
+				endpointmanager.LinkContainerID(ep)
```
Comment (tgraf) [inline daemon/docker_watcher.go]: Calling `LinkContainerID` only makes sense after setting `ep.DockerID`, in this branch, we know that the endpoint exists (it was retrievable by endpoint ID but not by DockerID). I think we should keep the code to set `ep.DockerID`

File: daemon/docker_watcher.go [hunk 2]
```diff
@@ -258,35 +253,31 @@ func (d *Daemon) handleCreateContainer(id string, retry bool) {
...
-			}
-		} else {
-			cont = container.NewContainer(dockerContainer, lbls)
-			cont.Mutex.Lock()
+		if !ok {
+			cont = container.NewContainer(dockerContainer)
 		}
 
+		ep.Mutex.Lock()
+		ep.DockerID = id
```
Comment (tgraf) [inline daemon/docker_watcher.go:262]: Or we can call `endpointmanager.LinkContainerID(ep)` here.

Comment (amreshakim) [reply daemon/docker_watcher.go:262]: good idea!

**Comment by amreshakim** [comment]
I messed up my branch on a forker repo so I will create another pull request

**Pull request closed without merging**